images/base/files/etc/sysctl.d/10-network-magic.conf

```
@@ -9,3 +9,17 @@
 # Kubernetes mitigates the possible security issue caused by enabling this option.
 # ref: https://nvd.nist.gov/vuln/detail/CVE-2020-8558
 net.ipv4.conf.all.route_localnet=1
+
+# The global kernel parameter net.ipv4.conf.all.arp_ignore governs the
+# conditions under which ARP requests will be accepted or ignored. This global
+# setting will override any individual interface settings. Some host systems
+# might set this global parameter to a more restrictive setting of 2 (or
+# greater). Specifically, in mode 2, the system ignores ARP requests directed to
+# /32 addresses (this is what kindnet assigns to its veth interfaces) because
+# the request originates from a different subnet. This behavior breaks routing
+# for kindnet (specifically the ptp plugin), so we opinionatedly set this to 1
+# when kindnet is in use to ensure proper networking.
+#
+# For more information see:
+# https://www.kernel.org/doc/html/latest/networking/ip-sysctl.html#arp_ignore
+net.ipv4.conf.all.arp_ignore=1
```

```suggestion
net.ipv4.conf.all.arp_ignore=0
```

default is 0 https://sysctl-explorer.net/net/ipv4/arp_ignore/ , 1 seems to break some scenarios https://serverfault.com/questions/935366/why-does-arp-ignore-1-break-arp-on-pointopoint-interfaces-kvm-guest, I think is better to use 0

---

I mentioned in the issue that in my testing 1 seemed sufficient, and all the CI seems to pass with this. I suspect that 1 may actually be reasonable given that I don't think we need to be able to respond to ARP requests for the addresses for other interfaces, but at the same time "0" being the historical setting I can see wanting to start there first. Changed in 5e66ec05a5fb40f41c7e466bd24d3e248ecc1efc